test/main/deepEqual.js

```diff
@@ -1581,6 +1581,35 @@ QUnit.test( "Test that must be done at the end because they extend some primitiv
 	}
 );
 
+QUnit.test( "Compare primitive objects", function( assert ) {
+	var SafeNumber = Number,
+	SafeString = String,
+	SafeBoolean = Boolean;
+
+	assert.equal( QUnit.equiv(new SafeNumber(1), new SafeNumber(1) ), true,
+		"Number: When create same parameter should be equal."
+	);
+	assert.equal( QUnit.equiv(new SafeNumber(1), new SafeNumber(2) ), false,
+		"Number: When create different parameter should be not equal."
```

Comment:
Maybe replace the assertion message with:

"Number objects with different values are not equivalent" or "QUnit.equiv returns false for Number objects with different values"